src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -752,13 +753,19 @@ private final static boolean isVCFStream(final InputStream stream, final String
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
-    private final String[] INT_DECODE_ARRAY = new String[10000];
+
     private final int[] decodeInts(final String string) {
-        final int nValues = ParsingUtils.split(string, INT_DECODE_ARRAY, ',');
+        // find number of CSV items to set array size
+        int numItems = string.split(",").length;
```

Comment:
Yes, both would have to parse the entire string but countMatches() avoids the construction and copy into the array of strings. I am having trouble accessing org.apache.commons.lang.StringUtils (which contains countMatches()) instead of com.sun.deploy.util.StringUtils. Looks like a classpath problem. Looked in pom.xml and found org.apache.commons dependency...